fix(products): Enhance tooltips and documentation (#134)
ni/systemlink-grafana-plugins: src/datasources/products/query_help.md

> The _SystemLink Products_ data source allows you to display product metadata in dashboards.

Reviewer: @richie-ni Since the existing SystemLink data sources have very less documentation. I was looking for some standard Grafana data sources for inspiration and found `Infinity` and `Redis`.

You can try them out here - https://play.grafana.org/d/infinity/2dc7103?orgId=1&from=now-24h&to=now&timezone=utc&editPanel=24

> The _SystemLink Products_ data source allows you to display product metadata in dashboards.

Reviewer: Can we add a title to the help content - `SystemLink Products data source`

> The _SystemLink Products_ data source allows you visualize metadata and gain insights related to your products.

Reviewer 2: ```suggestion
The _SystemLink Products_ data source allows you visualize and gain insights from the products.
```

I am not sure about using phrases like "your product" will be appropriate here, it can be little more generic. Ultimately, the decision should be from POs @sudharsan-ni

> The _SystemLink Products_ data source allows you visualize metadata and gain insights related to your products.

Reviewer: ```suggestion
The _SystemLink Products_ data source allows you to display product properties in dashboards and derive insights.
```
We can be consistent with usage of `properties` as discussed in HLD or initial AC PR, I guess.

### Key Features

> - **Metadata Visualization**: Display various metadata attributes of your products, such as part numbers, names, families, and workspaces, directly in your dashboards.

Reviewer: ```suggestion
- **Display product properties**: Display various properties of your products, such as part number, product name, family, and workspace, directly in your dashboards.
```

> - **Custom Queries**: Create custom queries to filter and sort product metadata based on specific criteria, allowing you to focus on the most relevant information.

Reviewer: ```suggestion
- **Custom Queries**: Create custom queries to filter and sort product properties based on specific query conditions, allowing you to get the most relevant product information.
```

### Example Queries

Here are some example queries you can create using the _SystemLink Products_ data source:

> - **List All Products**: Retrieve a list of all products along with their metadata attributes.

Reviewer: ```suggestion
- **List all products under a family**: Get the list of all products under a family along with their properties.
```
